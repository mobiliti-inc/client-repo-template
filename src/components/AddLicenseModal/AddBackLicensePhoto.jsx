import React from 'react';
import AddLicensePhotoModal from './AddPhotoModal';

class AddBackLicensePhoto extends React.PureComponent {
	static propTypes = {
		modalIsVisible: PropTypes.bool.isRequired,
		closeModal: PropTypes.func.isRequired,
		onBackLicensePhotoModalBack: PropTypes.func.isRequired,
		onUsePhoto: PropTypes.func.isRequired,
		closeCamera: PropTypes.bool.isRequired
	}
	render() {
		return (
			<AddLicensePhotoModal
				visible={this.props.modalIsVisible}
				title="Back of license"
				closeModal={this.props.closeModal}
				onBack={this.props.onBackLicensePhotoModalBack}
				onUsePhoto={this.props.onUsePhoto}
				closeCamera={this.props.closeCamera}
				onCameraAccessFail={() => { }}
			/>
		);
	}
}


export default CSSModules(AddBackLicensePhoto);
