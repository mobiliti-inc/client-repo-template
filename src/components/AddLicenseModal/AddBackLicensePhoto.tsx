import React, { PureComponent } from 'react';
import AddLicensePhotoModal from './AddPhotoModal';

type AddBackLicensePhotoProps = {
	modalIsVisible: boolean,
	closeModal: (...args: any[]) => any,
	onBackLicensePhotoModalBack: (...args: any[]) => any,
	onUsePhoto: (...args: any[]) => any,
	closeCamera: boolean
};

class AddBackLicensePhoto extends PureComponent<
	AddBackLicensePhotoProps,
	{}
	> {
	render() {
		return (
			<AddLicensePhotoModal
				visible={this.props.modalIsVisible}
				title="Back of license"
				closeModal={this.props.closeModal}
				onBack={this.props.onBackLicensePhotoModalBack}
				onUsePhoto={this.props.onUsePhoto}
				closeCamera={this.props.closeCamera}
				onCameraAccessFail={() => { }}
			/>
		);
	}
}

export default CSSModules(AddBackLicensePhoto);
