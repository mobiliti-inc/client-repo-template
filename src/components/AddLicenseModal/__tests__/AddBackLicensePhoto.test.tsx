import AddBackLicensePhoto from '../AddBackLicensePhoto';

describe('AddBackLicensePhoto', () => {
	it('renders', () => {
		const subject = shallow(
			<AddBackLicensePhoto
				modalIsVisible
				closeModal={() => { }}
				onBackLicensePhotoModalBack={() => { }}
				onUsePhoto={() => { }}
				closeCamera
				onCameraAccessFail={() => { }}
			/>);
		matchSnapshot(subject);
	});
});
