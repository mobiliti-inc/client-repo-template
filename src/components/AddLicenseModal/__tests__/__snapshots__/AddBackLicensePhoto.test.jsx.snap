// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBackLicensePhoto renders 1`] = `
<AddPhotoModal
  closeCamera={true}
  closeModal={[Function]}
  onBack={[Function]}
  onCameraAccessFail={[Function]}
  onUsePhoto={[Function]}
  title="Back of license"
  visible={true}
/>
`;
